patches/denymount+2.2.0.patch

```diff
@@ -0,0 +1,38 @@
+diff --git a/node_modules/denymount/lib/index.js b/node_modules/denymount/lib/index.js
+index 93b8e59..86d53dc 100644
+--- a/node_modules/denymount/lib/index.js
++++ b/node_modules/denymount/lib/index.js
+@@ -24,7 +24,9 @@ var utils = require('./utils');
+  * @module denymount
+  */
+ 
+-var EXECUTABLE_PATH = path.join(__dirname, '..', 'bin', 'denymount');
++var EXECUTABLE_PATH = process.pkg
++  ? path.join(path.dirname(process.execPath), 'denymount')
++  : path.join(__dirname, '..', 'bin', 'denymount');
```

Comment:
As far as my investigation went... The pkg standalone installer bundles together the node.js executable with the node_modules folder, all in a single large executable file. But it seems that a pkg limitation means that the OS cannot run executable binaries inside the large executable binary. (I guess this is the same limitation that requires native/binary modules to be packaged outside the large executable.) I had success copying the `denymount` executable to the `build-bin` folder, but the `node_modules/denymount/bin/denymount` location was hardcoded in the denymount `index.js` file. So I patched the file location using the `patch-package` npm package.

Reply:
@pdcastro denymount is our own (in fact, if memory serves, I think I did the initial implementation for Etcher :) — can’t this change be integrated directly? Repo is here: https://github.com/balena-io-modules/denymount

Reply:
@dfunckt, thanks for the review! Indeed I had missed the fact that 'denymount' is our own, and had initially used `patch-package` exactly because I wanted to avoid having to wait for the package's maintainer to accept a PR. Following your review, I created a `denymount` PR that would probably solve the problem more cleanly by avoiding the need for `patch-package`: https://github.com/balena-io-modules/denymount/pull/8
However, what I then found in the `denymount` repo/PR is that even the head of master (before any change of mine) would not pass the existing tests on Linux (they pass on macOS, but not on Linux), and the balena CI tests would thus fail. I think the tests require some changes to Sinon mocking. Unfortunately, following a recent priority discussion with Shaun, I must focus on CLI native installers and I cannot justify spending more time on the `denymount` PR. Some options now:
* We could merge this CLI PR as it is, as it solves the problem and users are waiting for the fix (#1212, #977).
* If you are interested, as the original author of the `denymount` code (?), you could take over that PR (https://github.com/balena-io-modules/denymount/pull/8). If you did that, I already have an alternative CLI PR that passes the `executablePath` option in the call to `denymount`:
```
# Coffeescript code
    dmOpts = {}
    if process.pkg
        dmOpts.executablePath = path.join(path.dirname(process.execPath), 'denymount')
    denymount params.target, dmHandler, dmOpts
```
* We could do both things: merge this PR now, and improve the solution later.

Reply:
@pdcastro I think the best option is to merge this one as-is, I don't see myself picking that up soon unfortunately.

Reply:
@dfunckt that's fine, then all I need is your review approval so I can click the green merge button. :-)

Reply:
@pdcastro those Linux checks are a relic of resin-ci going wild, they don't actually have to pass because they don't even make sense (it's a project specific to the Mac afterall, it doesn't compile anywhere else) and they're not required. All you care about are those travis-ci checks. So it still seems to be an option to update denymount and refer to it, but it's really up to you.

Reply:
I didn't see your last comment. I've approved the other PR, and I'll also approve this one so you can go either way at will.

Reply:
Ha - I hadn't seen your comment either :-) (about the `denymount` resin-ci tests not being required), only saw it after I merged this PR. I've now merged the `denymount` PR as well, as it's not a breaking change. I will update the CLI in a future PR.